### PR TITLE
Fix built in JPEG resolution parsing.

### PR DESCRIPTION
**Pull request by whyman**
Looks like it was broken with the std::byte changes.
Added a test to make fixing the bug easier.
Added Resolution object and tests
Various related tidy ups.

Fixes: https://github.com/gerbera/gerbera/issues/2595

### CODE REVIEW

File: src/metadata/resolution.cc [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    You should have received a copy of the GNU General Public License
+    along with Gerbera.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/// \file resolution.cc
+
+#include "resolution.h"
+#include "util/tools.h"
+
+Resolution::Resolution(std::string string)
```
Comment (neheb) [inline src/metadata/resolution.cc:27]: const string ref

**Comment by neheb** [comment]
```
 /home/runner/.conan/data/gerbera/dev/_/_/build/d9eb8116466780ac9e52023ad71eb294759c4452/src/util/jpeg_resolution.cc:124:17: error: comparison of integer expressions of different signedness: ‘long unsigned int’ and ‘int’ [-Werror=sign-compare]
  124 |         if (got != itemLen - 2) {
```

File: web/js/jquery.gerbera.editor.js [hunk 2]
```diff
@@ -162,7 +162,7 @@
     if (item) {
       reset(modal);
       if (item.image) {
-        modal.find('#mediaimage').prop('src', item.image.value);
+        modal.find('#mediaimage').prop('src', '//' + item.image.value);
```
Comment (KarlStraussberger) [inline web/js/jquery.gerbera.editor.js:165]: What's the background?

`image` is always the full path starting with virtual Url?

Comment (whyman) [reply web/js/jquery.gerbera.editor.js:165]: Yes, but Firefox refused to load it without a protocol element (http://) so I used the "//" which means match the pages protocol

Comment (KarlStraussberger) [reply web/js/jquery.gerbera.editor.js:165]: What's your virtualUrl setting in config?

Comment (whyman) [reply web/js/jquery.gerbera.editor.js:165]: Good question, I'm on holiday for a few days

Comment (KarlStraussberger) [reply web/js/jquery.gerbera.editor.js:165]: Enjoy it!

Comment (KarlStraussberger) [reply web/js/jquery.gerbera.editor.js:165]: Did you check you config. The change breaks my WebUI

Comment (whyman) [reply web/js/jquery.gerbera.editor.js:165]: So, I have virtualURL as `hostname:port` in config.

I guess you have `http://hostname:port` which would have broken.

I will put together a PR to fix it.

**Pull request merged**